platformtests/aws/default_instance_class_test.go

```diff
@@ -164,17 +164,16 @@ func TestGetDefaultInstanceClass(t *testing.T) {
 				}
 
 				if reflect.DeepEqual(available[instanceClass], zones) {
-					match = instanceClass
-					break
+					allowed = append(allowed, instanceClass)
 				}
 			}
 
-			if match == "" {
+			if len(allowed) == 0 {
 				t.Fatalf("none of the preferred instance classes are fully supported: %v", available)
 			}
 
 			t.Log(available)
-			assert.Equal(t, defaults.InstanceClass(region), match)
+			assert.Contains(t, allowed, defaults.InstanceClass(region))
```

Comment:
With this change we no longer warn when our default does not match the calculated preferred class.  That's appropriate for ap-northeast-2, but if, for example, one of the other regions happened to gain full support for m4, we'd want to switch our default back to m4 and wouldn't have the tests suggesting that change.  It's probably not a big deal as long as we prefer the older version, but when we start [preferring m5 over m4][1] it will become more important for us to track the m5 rollout across AWS regions.

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1710981